tests/test_select.py

```diff
@@ -37,17 +39,31 @@ def test_select_by_topic_filters_articles():
         ),
     ]
 
+    topic_filter = TopicFilter()
+    sampler = UniformSampler(num_slots=2)
+
+    pipeline = RecommendationPipeline(name="random_topical")
+    pipeline.add(topic_filter, inputs=["candidate", "profile"], output="topical")
+    pipeline.add(sampler, inputs=["topical", "candidate"], output="recs")
+
     # If we can, only select articles matching interests
-    recs = select_by_topic(articles, profile, num_slots=2)
+    inputs = {
+        "candidate": ArticleSet(articles=articles),
+        "clicked": ArticleSet(articles=[]),
+        "profile": profile,
+    }
+    recs = pipeline(inputs)
```

Comment:
The code above is repeated from `default.py`, which we'd ideally avoid by either restructuring the construction of recommender pipelines to make it re-usable or by reading the pipeline configuration from a file.